src/tir/transforms/lower_sparse_buffer.cc

```diff
@@ -354,6 +354,76 @@ class BufferTransformer : public StmtExprMutator {
   bool is_horizontal_fuse_;
 };
 
+class Sparse_indices_buffer_post_process : public StmtExprMutator {
+  public:
+    explicit Sparse_indices_buffer_post_process(){}
+  private:
+    Stmt VisitStmt_(const BlockNode* op) final{
+      if(support::StartsWith(op->name_hint,"binary_search_block"))
+        skip=true;
+      auto ret = StmtExprMutator::VisitStmt_(op);
+      skip=false;
+      return ret;
+    }
+
+    PrimExpr VisitExpr_(const BufferLoadNode* op) final{
+      if(!skip&&support::StartsWith(op->buffer->name,"mid_")&&buffer_processed.count(op->buffer)==0)
+      {
+        buffer_need_process.push_back(GetRef<BufferLoad>(op));
+        find++;
+        return PrimExpr(0);
+      }
+      else
+      {
+        auto find_backup=find;
+        for(auto i : op->indices)
+        {
+          VisitExpr(i);
+        }
+        if(find!=find_backup)
+        {
+          find=find_backup;
+          if(op->buffer->dtype.is_float())
+            return PrimExpr(FloatImm(DataType::Float(32), 0));
+          else if(op->buffer->dtype.is_int())
+            return PrimExpr(FloatImm(DataType::Int(32), 0));
+          else
+            throw tvm::Error("sparse tensor should only be int or float");
+        }
+      }
+      return StmtExprMutator::VisitExpr_(op);
+    }
+    
+    Stmt VisitStmt_(const BufferStoreNode* op) final {
+      int original_size = buffer_need_process.size();
+      PrimExpr value = VisitExpr(op->value);
+      if(original_size<buffer_need_process.size())
+      {
+        if(original_size+1!=buffer_need_process.size())
+        {
+          auto bufferload = buffer_need_process[original_size+1];
+          for(int i=original_size+2;i<buffer_need_process.size();i++)
+          {
+            CHECK(buffer_need_process[i].same_as(bufferload))<<"current only allow same mid buffer load expr in one buffer store stmt";
+          }
+        }
+        auto buffer_found = buffer_need_process.back();
+        buffer_need_process.erase(buffer_need_process.end()-1);
+        buffer_processed.insert(buffer_found->buffer);
+        PrimExpr if_stmt = (-1 != buffer_found);
+        auto new_stmt=IfThenElse(if_stmt,StmtExprMutator::VisitStmt_(op),BufferStore(op->buffer,value,op->indices,op->span));
+        buffer_processed.erase(buffer_found->buffer);
+        return new_stmt;
+      }
+      else
+      return StmtExprMutator::VisitStmt_(op);
+    }
+    std::vector<BufferLoad> buffer_need_process;
+    std::set<Buffer> buffer_processed;
```

Comment:
Use `std::set<const BufferNode*>` instead, otherwise, it's not clear which compare function will be used to determnie (actually we only want to compare pointers).